### PR TITLE
add: zed-nightly

### LINKED ISSUE #1011: Inclusion of x86_64-lg experimental CI runners
This might fixes build errors for #922 (GitHub runners are not capable at doing such resource-consuming compilations)

Just need to edit our CI workflows (I hope)

### PR DESCRIPTION
**Pull request by june-fish**

### CODE REVIEW

**Review by lleyton — DISMISSED** [review]

File: anda/devs/zed-nightly/update.rhai [hunk 1]
```diff
@@ -0,0 +1,9 @@
+if filters.contains("nightly") {
+  rpm.global("commit", gh_commit("zed-industries/zed"));
+  if rpm.changed() {
+    let v = find("(\\d+\\.\\d+\\d+\\.\\d+)", find("\nversion = \"(\\d+\\.\\d+\\d+\\.\\d+)\"\n", gh_rawfile("zed-industries/zed", "main", "crates/zed/Cargo.toml"), 1), 1);
```
Comment (madonuko) [inline anda/devs/zed-nightly/update.rhai]: Is this double nested find() collapsible?

File: anda/devs/zed-nightly/zed-nightly.spec [hunk 2]
```diff
@@ -0,0 +1,95 @@
+%global commit 0129d4e2506d5ec5e50ef0968382770b9abec390
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+%global commit_date 20240619
+%global ver v0.142.0
```
Comment (madonuko) [inline anda/devs/zed-nightly/zed-nightly.spec]: Strip the `v` prefix as according to update.rhai

File: anda/devs/zed-nightly/zed-nightly.spec [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+BuildRequires:  perl-File-Copy
+BuildRequires:  perl-lib
+BuildRequires:  vulkan-loader
+
+%description
+Code at the speed of thought - Zed is a high-performance, multiplayer code editor from the creators of Atom and Tree-sitter.
+
+%prep
+%autosetup -n %{crate}-%{commit} -p1
+%cargo_prep_online
```
Comment (madonuko) [inline anda/devs/zed-nightly/zed-nightly.spec]: Useless?

**Review by madonuko — CHANGES_REQUESTED** [review]

File: anda/devs/zed-nightly/update.rhai (hunk 1 above)
Comment (june-fish) [reply anda/devs/zed-nightly/update.rhai]: I don't think so.

File: anda/devs/zed-nightly/zed-nightly.spec (hunk 2 above)
Comment (june-fish) [reply anda/devs/zed-nightly/zed-nightly.spec]: Done

File: anda/devs/zed-nightly/zed-nightly.spec (hunk 3 above)
Comment (june-fish) [reply anda/devs/zed-nightly/zed-nightly.spec]: Removed

File: anda/devs/zed-nightly/update.rhai (hunk 1 above)
Comment (madonuko) [reply anda/devs/zed-nightly/update.rhai]: the outer find() doesn't actually do anything I think?

File: anda/devs/zed-nightly/zed-nightly.spec [hunk 4]
```diff
@@ -0,0 +1,95 @@
+%global commit 0129d4e2506d5ec5e50ef0968382770b9abec390
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+%global commit_date 20240619
+%global ver 0.142.0
+
+%bcond_without check
+%global debug_package %{nil}
```
Comment (madonuko) [inline anda/devs/zed-nightly/zed-nightly.spec]: Not needed

**Review by madonuko — CHANGES_REQUESTED** [review]

**Comment by lleyton** [comment]
This is ready for merge AFTER TALKING TO ZED, the x86_64 test build failed due to an intermittent runner issue @madonuko feel free to approve it

**Review by madonuko — DISMISSED** [review]

**Review by lleyton — APPROVED** [review]

**Comment by raboneko** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|f39|[](https://github.com/terrapkg/packages/pull/1394)|
|✅|frawhide|[](https://github.com/terrapkg/packages/pull/1395)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/9690533487) for details

**Pull request merged**